### DISCUSSION TITLE
Javascript Error only on initial value

### DISCUSSION THREAD

**Discussion by ttelik** [Budibase/budibase#4319] [Help] [answered]
Hi Everyone,

I'm experimenting with Budibase for the first time and am very impressed with the ease of use. 

One snag that I'm having occurs when I use a javascript function to set the default value of some form fields. This function is taking data from one form field and parsing it to pre-fill another. The JS is set up such that it will return an empty string if the source field is blank or NULL.

The behavior is very strange. Upon page load I get a JS error message in the field. Once I type text into the source field (MODEL) in this case, the JS works as intended. Also, if I clear the source field, the JS also works as intended (the target field is blank, no error).

Is there some condition I need to catch in my code in order to prevent the unsightly error from appearing upon page load? From a purely functional standpoint, the code works just fine.

Here's an example of a JS function to set a default value for a field:
```
let rawModel = $("Production Data Form.Fields.MODEL");

if ( rawModel === "" || rawModel == NULL ) {
	return "";
	
} else if ( rawModel.includes("-") ) {
	
	model = rawModel.split("-");
	return model[0];
	
} else {
	return "";
	
}
```

**Comment by pavleks** [comment, marked as answer]
Hi @ttelik please add `|| rawModel == undefined` or just change from NULL to undefined. I just tested it and looks like  JS error message in the field disappeared.

**Reply by ttelik** [reply]
Perfect, thanks!
